Hardware/CMockHardwareMap.cpp

```diff
@@ -238,16 +238,16 @@ bool CMockHardwareMap::newCommand(ICommand *p_command,
     {
         switch (result)
         {
-            case COMMAND_OK:
+            case ICommand::COMMAND_OK:
                 p_comchannel->send("OK.\n");
                 break;
-            case ERROR_ARG_COUNT:
+            case ICommand::ERROR_ARG_COUNT:
```

Comment:
Wouldn't it be cleaner to send a string version of the error enum "ERROR_ARG_COUNT" or "ERROR_OUT_OF_BOUNDS" etc to keep up with consistency in other places where error messages may also be sent?

Reply:
Don't follow the comment.

Reply:
Since this enum is part of ICommand interface class, other classes will implement this interface to process commands with their own custom message.  Over time this will become difficult to manage since each implementation will have its own unique & hardcoded message also since the message holds the same amount of information as the enum name it makes sense to send a string representation of the ENUM name. This is however not supported by default in C & C++ (unlike python) nevertheless, there are solutions which I think are neat. 

Proposed Solution C & C++
[Example C & C++](https://godbolt.org/z/44hqEef6o )


`
#include <iostream>
#include "magic_enum.hpp" // Header only libarary 

using namespace std;

/*************** C Implementation - Manual Management *************/
enum ErrorType { COMMAND_OK = 0, ERROR_OUT_OF_BOUNDS, ERROR_TYPE_MISMATCH, MAX_ERROR_TYPE}; 

static const char * ErrorTypeName [MAX_ERROR_TYPE] = {
    /* ENUM -> STRING */
    [COMMAND_OK] = "COMMAND_OK",
    [ERROR_OUT_OF_BOUNDS] = "ERROR_OUT_OF_BOUNDS",
    [ERROR_TYPE_MISMATCH] = "ERROR_TYPE_MISMATCH"
};


int main()
{
    /** C Like **/
    cout << "C String Implementation " << ErrorTypeName[COMMAND_OK] << endl;
    
    /** C++ 17 Magic Enum Implementation **/
    cout << "C++ 17 Magic Enum Implementation " << magic_enum::enum_name(ERROR_OUT_OF_BOUNDS);
    return 0;
}
`

Reply:
What if I want to send a speciall message? Do I create an entry in the dictionary for sake of single use? I generally agree that if we had a large somewhat disparate collection of users who where messages might get out of control if left unchecked, proposed solution would be suitable. However, we will only have at most 3-4 controllers, so creating another layer of complexity to solve a problem that'll likely not exist, is poor use of resources. Let's not optimise something that works.